src/utils/logger.ts

```diff
@@ -1,4 +1,219 @@
+
+export interface LoggerInterface {
   log: (...args: any[]) => void
   error: (...args: any[]) => void
-}
\ No newline at end of file
+  info: (...args: any[]) => void
+}
+
+export type ErrorMessage = {
+  errorMessage: string
+  stackTrace?: string
+}
+
+export type Log = {
+  custom: string
+  traceId?: string
+  level?: string
+  storeId?: string
+  email?: string
+  role?: string
+  http?: {
+    method?: string
+    version?: string
+    url?: string
+    protocol?: string
+    srcIp?: string
+    statusCode?: number
+    clientDevice?: string
+    contentLength?: number
+  }
+  consumer: {
+    topic: string
+  }
+  reqBody?: string
+  errorMessage?: string
+  stackTrace?: string
+  responseTime?: number
+}
+
+export type LogType = 'http' | 'sqs'
+export type LogLevel = 'INFO' | 'ERROR' | 'DEBUG'
+export class Logger implements LoggerInterface {
+  private startMs: number
+  private logObject: Log
+  private service: string
+  private logType: LogType
+  private logFilterKey: string
+  private enabled: boolean
+
+  constructor(
+    logFilterKey: string,
+    logType: LogType,
+    service: string,
+    enabled: boolean
+  ) {
+    this.startMs = new Date().getTime()
+    this.service = service || '-'
+    this.logType = logType
+    this.logFilterKey = logFilterKey
+    this.enabled = enabled
+  }
+
+  /**
+   * Query elapsed milliseconds.
+   *
+   * @return milliseconds
+   */
+  private get elapsedMs(): number {
+    return new Date().getTime() - this.startMs
+  }
+
+  private instanceOfKolpServiceContext(
+    object: any
+  ): object is KolpServiceContext {
+    return object.discriminator === 'context'
+  }
+
+  public log(o: any) {
+    console.log(o)
+  }
+
+  public info(ctx: KolpServiceContext | SQSRecord) {
+    if (!this.enabled) return
+    this.instanceOfKolpServiceContext(ctx)
+      ? this.parseHTTPContextToLogFormat('INFO', ctx)
+      : this.parseSQSContextToLogFormat('INFO', ctx)
+  }
+  public error(
+    ctx: KolpServiceContext | SQSRecord,
+    errorMessage: ErrorMessage
+  ) {
+    if (!this.enabled) return
+    if (this.instanceOfKolpServiceContext(ctx)) {
+      this.parseHTTPContextToLogFormat('ERROR', ctx, errorMessage)
+    } else {
+      this.parseSQSContextToLogFormat('ERROR', ctx, errorMessage)
+    }
+  }
+
+  private detemineErrorMessage(errorMessage: ErrorMessage) {
+    return errorMessage
+      ? {
+          errorMessage: errorMessage.errorMessage,
+          stackTrace: errorMessage.stackTrace,
+        }
+      : {
+          errorMessage: '-',
+          stackTrace: '-',
+        }
+  }
+
+  private parseHTTPContextToLogFormat(
+    level: LogLevel,
+    ctx: KolpServiceContext,
+    errorMessage?: ErrorMessage
+  ): void {
+    this.writeLog(level, {
+      custom: this.logFilterKey,
+      level: level,
+      traceId:
+        ctx.traceId ||
+        `${ctx.req.method}-${ctx.originalUrl}-${ctx.event.requestContext.requestId}`,
+      storeId: ctx.params.storeId || 'shared',
```

Comment:
this is not the context of kolp. These context should be used in internal project.

Kolp should be much more generic.